srml/bridge/src/lib.rs

```diff
@@ -241,14 +240,19 @@ where
 	B: BlockT<Hash=H256>,
 	NumberFor<B>: grandpa::BlockNumberOps,
 {
-	let grandpa_justification = GrandpaJustification::<B>::decode_and_verify_finalizes(
+	// We don't really care about the justification, as long as it's valid
+	let j = GrandpaJustification::<B>::decode_and_verify_finalizes(
 		&justification,
 		(hash, number),
 		set_id,
 		voters,
-	).unwrap();
+	);
 
-	Ok(())
+	// Uh, there's got to be a nicer way of doing this...
+	match j {
```

Comment:
you could do (provided that your error type implement client::Error -> Error):

```
	GrandpaJustification::<B>::decode_and_verify_finalizes(
		&justification,
		(hash, number),
		set_id,
		voters,
	)?;

	Ok(())
```

Reply:
Ideally yes, but the error types don't implement that :P. I don't really want to implement it since at the end of the day I don't actually care about the error that comes from `decode_and_verify_finalizes`, I just want to know that an error happened.

Reply:
ok, then I guess you could chain `map(|_| ()).map_err(|_| Error::InvalidFinalityProof)` but it is almost uglier :P